BlueprintUI/Sources/Focus/FocusState.swift

```diff
@@ -0,0 +1,238 @@
+/// proceeds.
+///
+///     struct LoginForm: ProxyElement {
+///         enum Field: Hashable {
+///             case username
+///             case password
+///         }
+///
+///         var username: String
+///         var password: String
+///         var handleLogin: () -> Void
+///
+///         @FocusState private var focusedField: Field?
+///
+///         var elementRepresentation: Element {
+///             Column { column in
+///                 column.add(
+///                     child: TextField(text: "")
+///                         .focused(when: $focusedField, equals: .username)
+///                 )
+///
+///                 column.add(
+///                     child: TextField(text: "")
+///                         .focused(when: $focusedField, equals: .password)
+///                 )
+///
+///                 column.add(
+///                     child: Button(
+///                         onTap: {
+///                             if username.isEmpty {
+///                                 focusedField = .username
+///                             } else if password.isEmpty {
+///                                 focusedField = .password
+///                             } else {
+///                                 handleLogin()
+///                             }
+///                         },
+///                         wrapping: Label(text: "Sign In")
+///                     )
+///                 )
+///             }
+///         }
+///     }
+///
+/// To allow for cases where focus is completely absent from a view tree, the wrapped value must be
+/// either an optional or a Boolean. Set the focus binding to `false` or `nil` as appropriate to
+/// remove focus from all bound fields. You can also use this to remove focus from a ``TextField``
+/// and thereby dismiss the keyboard.
+///
+/// ### Avoid Ambiguous Focus Bindings
+///
+/// A `TextField` can have only one focus binding, stored in its `focusBinding` property. If you apply
+/// the `focused` modifier multiple times, the last one will overwrite the previous value.
+///
+/// On the other hand, binding the same value to two views is ambiguous. In the following example,
+/// two separate fields bind focus to the `name` value:
+///
+///     struct Content: ProxyElement {
+///         enum Field: Hashable {
+///             case name
+///             case fullName
+///         }
+///
+///         @FocusState private var focusedField: Field?
+///
+///         var elementRepresentation: Element {
+///             Column { column in
+///                 column.add(
+///                     child: TextField(text: "")
+///                         .focused(when: $focusedField, equals: .name)
+///                 )
+///
+///                 column.add(
+///                     child: TextField(text: "")
+///                         .focused(when: $focusedField, equals: .name) // incorrect re-use of .name
+///                 )
+///             }
```

Comment:
Might be nice to show an example of how to give something focus `onAppear`

Reply:
> Might be nice to show an example of how to give something focused `onAppear`

Sure thing. I adapted these examples from [Apple's](https://developer.apple.com/documentation/swiftui/focusstate).

Reply:
added!